### PR TITLE
Add two new API functions: jerry_create_number_infinity and jerry_create_number_nan

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by martijnthe** [comment]
Thanks!

File: tests/unit/test-api.c [hunk 1]
```diff
@@ -700,9 +701,25 @@ main (void)
...
+  jerry_release_value (val_t);
+
+  val_t = jerry_create_number_nan ();
+  number_val = jerry_get_number_value (val_t);
+  TEST_ASSERT (number_val != number_val);
+  jerry_release_value (val_t);
+
+  val_t = jerry_create_number_infinity (true);
+  number_val = jerry_get_number_value (val_t);
+  TEST_ASSERT (number_val * 3 == number_val);
```
Comment (akosthekiss) [inline tests/unit/test-api.c]: In the unlikely event of `number_val == 0.0`, the assert would still pass, which is not what we want. Thus, I'd rewrite this checks as
```c
TEST_ASSERT (number_val * 3 == number_val && number_val != 0.0);
```

**Comment by akosthekiss** [comment]
After the assert in the api test is fixed: LGTM

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -1758,6 +1758,77 @@ jerry_create_number (double value);
...
 - [jerry_release_value](#jerry_release_value)
+- [jerry_create_number_infinity](#jerry_create_number_infinity)
+- [jerry_create_number_nan](#jerry_create_number_nan)
+
+
+## jerry_create_number_infinity
+
+**Summary**
+
+Creates a `jerry_value_t` representing a postive or negative infinity value.
```
Comment (bzsolt) [inline docs/02.API-REFERENCE.md]: Typo: `postive` -> `positive`

File: jerry-core/jerry.c [hunk 3]
```diff
@@ -847,6 +847,39 @@ jerry_create_number (double value) /**< double value from which a jerry_value_t
 } /* jerry_create_number */
 
 /**
+ * Creates a jerry_value_t representing a postive or negative infinity value.
```
Comment (bzsolt) [inline jerry-core/jerry.c]: Typo: `postive` -> `positive`

**Comment by LaszloLango** [comment]
LGTM

**Pull request merged**